packages/react/src/components/Authenticator/SignUp/SignUp.tsx

```diff
@@ -33,11 +35,12 @@ export function SignUp() {
   const [_state, send] = useAuth();
   const actorState: SignUpState = getActorState(_state);
   const isPending = actorState.matches('signUp.pending');
-  const { remoteError } = actorState.context;
   const { validationError } = getActorContext(_state) as SignUpContext;
 
-  const [primaryAlias, ...secondaryAliases] = _state.context.config
-    ?.login_mechanisms ?? ['username', 'email', 'phone_number'];
+  const [primaryAlias, ...secondaryAliases] =
+    _state.context.config?.login_mechanisms?.filter(
+      (alias: any): alias is UserNameAlias => userNameAliasArray.includes(alias)
+    ) ?? userNameAliasArray;
```

Comment:
Let's clean this up. `alias` shouldn't be `any` - either `string` or `LoginMechanism`, right? We also don't really need the `alias is UserNameAlias` inside of the filter there.

```
_state.context.config?.login_mechanisms?.filter(
  (alias: LoginMechanism) => userNameAliasArray.includes(alias)
) ?? userNameAliasArray;
```

Reply:
Yeah the `any` typing for `alias` isn't ideal and I just did it to get around how TypeScript types the `.includes` method. I'm essentially running into this [problem](https://github.com/microsoft/TypeScript/issues/26255), where `LoginMechanism` isn't a valid search value in `userNameAliasArray` because it includes the social providers. I'm open to other suggestions here, I mostly just wanted to limit how often we're writing `['username', 'email', 'phone_number']`.

The `alias is UserNameAlias` is nice because we're saying that `primaryAlias` will be `UserNameAlias` and `secondaryAliases` will be `UserNameAlias[]` after the filter is complete.

Reply:
Hm ok. I'll approve for now since playing with TypeScript isn't worth blocking over, but this might be something to revisit moving forward then.

Reply:
I captured this as an E2 task https://app.asana.com/0/1200334286823427/1201001812784243. Each `alias` begins as being a `LoginMechanism`, so I think revisiting our typings can get us to removing the `any` type to get around this issue.